cli/src/main/java/org/jboss/as/cli/handlers/ShutdownHandler.java

```diff
@@ -244,6 +298,45 @@ protected boolean isLocalHost(ModelControllerClient client, String host) throws
         return result.asString().equals(host);
     }
 
+    protected boolean isLocalClientLaunch(CommandContext ctx) throws CommandLineException {
+        final ModelNode op = new ModelNode();
+        final ParsedCommandLine args = ctx.getParsedCommandLine();
+        final ModelControllerClient client = ctx.getModelControllerClient();
+        if(ctx.isDomainMode()) {
+            final String hostName = host.getValue(args);
+            if (hostName == null) {
+                throw new CommandFormatException("Missing required argument " + host.getFullName());
+            }
+            op.get(Util.ADDRESS).add(Util.HOST, hostName);
+            op.get(Util.ADDRESS).add(Util.CORE_SERVICE, "host-environment");
+
+        } else {
+            op.get(Util.ADDRESS).add(Util.CORE_SERVICE, "server-environment");
+        }
+        op.get(Util.OPERATION).set("query");
+        op.get("select").add(Util.HOME_DIR);
+
+        ModelNode response;
+        try {
+            response = client.execute(op);
+        } catch (IOException e) {
+            throw new CommandLineException("Failed to read attribute " + Util.HOME_DIR, e);
+        }
+        if(!Util.isSuccess(response)) {
+            throw new CommandLineException("Failed to read attribute " + Util.HOME_DIR
+                    + ": " + Util.getFailureDescription(response));
+        }
+        ModelNode result = response.get(Util.RESULT);
+        if(!result.isDefined()) {
+            throw new CommandLineException("The result is not defined for attribute " + Util.HOME_DIR + ": " + result);
+        }
+
+        final String jbossHome = WildFlySecurityManager.getEnvPropertyPrivileged("JBOSS_HOME", null);
+        return jbossHome != null && Paths.get(jbossHome).normalize().toAbsolutePath().equals(
```

Comment:
Is this check done to identify if the CLI and server processes are running inside the same host? Does the file path guarantees that?

Reply:
@jfdenise Resolve this if you are Ok with the approach